rust/blockstore/benches/writer.rs

```diff
@@ -0,0 +1,265 @@
+                data.sort_unstable_by(|a, b| a.0.cmp(&b.0).then_with(|| a.1.cmp(&b.1)));
+                data
+            },
+        ),
+    ];
+
+    {
+        let mut fresh_writer = c.benchmark_group(format!("fresh_writer_{}", name));
+        fresh_writer.throughput(criterion::Throughput::Bytes(data_byte_size as u64));
+
+        for (name, writer_options, data) in name_writer_options_data.iter() {
+            fresh_writer.bench_function(*name, |b| {
+                b.to_async(runner).iter_batched(
+                    || data.clone(),
+                    |data| async {
+                        let writer = provider
+                            .write::<D::Key, D::Value>(*writer_options)
+                            .await
+                            .unwrap();
+                        for (prefix, key, value) in data {
+                            writer.set(&prefix, key, value).await.unwrap();
+                        }
+
+                        writer.commit::<D::Key, D::Value>().await.unwrap();
+                    },
+                    criterion::BatchSize::LargeInput,
+                );
+            });
+        }
+    }
+
+    {
+        let populated_blockfile_id = runner.block_on(async {
+            let writer = provider
+                .write::<D::Key, D::Value>(BlockfileWriterOptions::new().unordered_mutations())
+                .await
+                .unwrap();
+
+            let generator = D::generate(size.clone());
+            for (prefix, key, value) in generator.data() {
+                writer.set(&prefix, key, value).await.unwrap();
+            }
+
+            let writer_id = writer.id();
+            let flusher = writer.commit::<D::Key, D::Value>().await.unwrap();
+            flusher.flush::<D::Key, D::Value>().await.unwrap();
+            writer_id
+        });
+
+        let mut forked_writer = c.benchmark_group(format!("forked_writer_{}", name));
+        forked_writer.throughput(criterion::Throughput::Bytes(data_byte_size as u64));
+
+        for (name, writer_options, data) in name_writer_options_data.iter() {
+            forked_writer.bench_function(*name, |b| {
+                b.to_async(runner).iter_batched(
+                    || data.clone(),
+                    |data| async {
+                        let writer = provider
+                            .write::<D::Key, D::Value>(writer_options.fork(populated_blockfile_id))
+                            .await
+                            .unwrap();
+
+                        for (prefix, key, value) in data {
+                            writer.set(&prefix, key, value).await.unwrap();
+                        }
+
+                        writer.commit::<D::Key, D::Value>().await.unwrap();
+                    },
+                    criterion::BatchSize::LargeInput,
+                );
+            });
+        }
+    }
+}
+
+// todo: this should be a parameter
+const BLOCK_SIZE: usize = 1024 * 256; // 256KB
```

Comment:
should our benchmark use our actual block size?

Reply:
yes, will update